### ISSUE TITLE
Feature Request: Collaborative Evaluation

### ISSUE THREAD

**Issue by AutoSponge** [w3c/wcag-em-report-tool#340]
As a member of an evaluation team or as a member of a community crowd-sourcing an evaluation, I would like the ability to work on the same, persisted document as other reviewers.

I'm eager to see if there's support for this feature request as well as the ideas that come out of a discussion.

**Comment by AutoSponge** [comment]
Idea 1:

## Scope
- Report Tool will accept a semantic "REST" endpoint as the "storage" option for a report.
- Report Tool will assume OAuth2 workflow for protected endpoints.
- Report Tool will communicate with the endpoint on "load" and "save" actions.
- Report Tool creates pre-conditions based on Etags (conflict resolution on write).

## Requirements
- Definition of the appropriate mime type(s): `application/earl+json` or `application/ld+json` etc.
- A spec (mostly referencing HTTP behaviors) for the server.
- Coding OAuth2 behavior into Report Tool client.
- Coding endpoint communication into Report Tool client.
- Coding sample spec-compliant server or appropriate mock for testing.

**Comment by juleskuehn** [comment]
I assume this wouldn't be part of the hosted functionality on w3.org but yes, our testing team is very interested in this as well.

**Comment by AutoSponge** [comment]
@juleskuehn Right. I assume the reference server or spec would provide a framework to build your own implementation from. As long as the server is compliant, any authorized user could theoretically use any client to add a report (or, more likely use the default client served by the server).

**Comment by rvantonisse** [comment]
> Report Tool will accept a semantic "REST" endpoint as the "storage" option for a report.

There is already something similar in place. It is handled by the [reportStorage service](https://github.com/w3c/wcag-em-report-tool/blob/master/app/scripts/services/reportStorage.js).

Essentially you need to add `autosave="true"` with an url `url="withyour.reporting-tool.com/api"` to the approot in the index.html to send and get data from the backend like this:

```html
<body ng-app="wcagReporter" autosave url="myReporter.com/report/1">
```

But the backend API to store the reportdata is something you would have to build by yourself.
